Sources/MapboxMaps/Foundation/Camera/BasicCameraAnimator.swift

```diff
@@ -47,139 +62,170 @@ public class BasicCameraAnimator: NSObject, CameraAnimator, CameraAnimatorInterf
 
     /// Add a completion block to the animator. 
     public func addCompletion(_ completion: @escaping AnimationCompletion) {
-        let wrappedCompletion = wrapCompletion(completion)
-        propertyAnimator.addCompletion(wrappedCompletion)
-    }
-
-    internal func wrapCompletion(_ completion: @escaping AnimationCompletion) -> (UIViewAnimatingPosition) -> Void {
-        return { [weak self] animationPosition in
-            guard let self = self else { return }
-            self.transition = nil // Clear out the transition being animated by this animator since the animation is complete if we are here.
-            self.delegate?.schedulePendingCompletion(forAnimator: self, completion: completion, animatingPosition: animationPosition)
-
-            // Invalidate the delayed animation timer if it exists
-            self.delayedAnimationTimer?.invalidate()
-            self.delayedAnimationTimer = nil
-        }
+        precondition(internalState != .final, "Attempt to add a completion block to an animation that has already completed.")
+        completions.append(completion)
     }
 
     /// Continue the animation with a timing parameter (`UITimingCurveProvider`) and duration factor (`CGFloat`).
     public func continueAnimation(withTimingParameters parameters: UITimingCurveProvider?, durationFactor: Double) {
-        propertyAnimator.continueAnimation(withTimingParameters: parameters, durationFactor: CGFloat(durationFactor))
+        switch internalState {
+        case .initial:
+            fatalError("Attempt to continue an animation that has not started.")
+        case .inProgress:
+            propertyAnimator.continueAnimation(withTimingParameters: parameters, durationFactor: CGFloat(durationFactor))
+        case .final:
+            precondition(internalState != .final, "Attempt to continue an animation that has already completed.")
+        }
     }
 
-    internal var currentCameraOptions: CameraOptions? {
-
-        // Only call jumpTo if this animator is currently "active" and there are known changes to animate.
-        guard propertyAnimator.state == .active,
-              let transition = transition else {
-            return nil
+    func update() {
+        switch internalState {
+        case .initial, .final:
+            return
+        case let .inProgress(transition):
+            // The animator has been started or paused, so get the interpolated value. This may be nil if
+            // the animations haven't yet propagated into the CameraView's presentation tree.
+            if let presentationCameraOptions = cameraView.presentationCameraOptions {
+                mapboxMap.setCamera(
+                    to: cameraOptions(
+                        with: transition,
+                        cameraViewCameraOptions: presentationCameraOptions))
+            }
         }
+    }
 
+    private func cameraOptions(with transition: CameraTransition, cameraViewCameraOptions: CameraOptions) -> CameraOptions {
```

Comment:
should we rename this method? This may be just me, but i'm not sure i fully understand what it's doing from the signature?

Alternatively, we could maybe add a comment here?

Reply:
Will add a comment